modules/ROOT/pages/provisioning-kubevirt.adoc

```diff
@@ -0,0 +1,87 @@
+= Provisioning Fedora CoreOS on KubeVirt
+
+This guide shows how to provision new Fedora CoreOS (FCOS) nodes on any KubeVirt-enabled Kubernetes cluster.
+
+== Prerequisites
+
+Before provisioning an FCOS machine, you must have an Ignition configuration file containing your customizations. If you do not have one, see xref:producing-ign.adoc[Producing an Ignition File].
+
+You also need to have access to a Kubernetes environment with https://kubevirt.io/user-guide/operations/installation/[KubeVirt] installed.
+
+== Referencing the KubeVirt Image
+
+Fedora CoreOS is designed to be updated automatically, with different schedules per stream.
+
+The image for each stream can directly be referenced from the official registry:
+
+----
+quay.io/coreos/fedora-coreos:{stable|next|testing}
+----
+
+== Launching a VM instance
+
+Now that you have an image it can be used in your VMI definitions. The following example will use the image from the stable stream.
+
+You'll also need the Ignition config.
+Here the Ignition config is exposed to the VMI via a Kubernetes Secret.
+Learn about various ways to expose userdata to VMIs in the https://kubevirt.io/user-guide/virtual_machines/startup_scripts/#startup-scripts[KubeVirt user guide].
+
+.Creating the secret
+[source, bash]
+----
+kubectl create secret generic ignition-payload --from-file=userdata=example.ign
+----
+
+.Launching a VM instance referencing the secret
+[source, bash]
+----
+cat << END > vmi.yaml
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: my-fcos
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+      rng: {}
+    resources:
+      requests:
+        memory: 1024M
+  volumes:
+  - containerDisk:
+      image: quay.io/coreos/fedora-coreos:stable
+    name: containerdisk
+  - name: cloudinitdisk
+    cloudInitConfigDrive:
+      secretRef:
+        name: ignition-payload
+END
+kubectl create -f vmi.yaml
+----
+
+Now you should be able to SSH into the instance. If you didn't change the defaults, the
+username is `core`.
+
+.Accessing the VM instance via ssh
+[source, bash]
+----
+kubectl virt ssh core@my-fcos
+----
+
+== Mirroring the image for use in private registries
+
+If a private registry in air-gaped installations is used, the image can be mirrored to that registry.
+
+.Mirroring a fcos image from stable stream
+[source, bash]
+----
+skopeo copy docker://quay.io/coreos/fedora-coreos:stable docker://myregistry.io/myorg/fedora-coreos:stable
+----
```

Comment:
may also be cool to document the "download from s3 and upload" case here too (using `coreos-installer download` I presume) in case people need to physically copy the file around to get it into the environment they want. The command above assumes skopeo has access to the net and to the disconnected env, which may not always be true.

Reply:
But one doesn't have to copy between two registries.  `skopeo copy docker://quay.io/coreos/fedora-coreos:stable oci:fcos` gives you a local directory that you can then copy around however you'd transport regular files (e.g. `cp fcos /mnt/usbstick`).  Then on the other end of the sneakernet, `skopeo copy oci:fcos docker://examplecorp.registry.int/fedora-coreos:stable`.

I think using native container copy flows should be preferred over plain HTTP.

Reply:
Ah, okay.  Let's document that flow explicitly then?

Reply:
Or to say it more strongly...I think we can just say "any tool that understands container images can be used to mirror", and link to e.g. https://github.com/containers/skopeo/blob/main/docs/skopeo-sync.1.md

(Which already documents exactly this with `--dir`, etc.)